mavros/src/lib/uas_data.cpp

```diff
@@ -47,13 +47,19 @@ UAS::UAS() :
 	catch (const std::exception &e) {
 		// catch exception and shutdown node
 		ROS_FATAL_STREAM("UAS: GeographicLib exception: " << e.what() <<
-				" | Run install_geographiclib_dataset.sh script in order to install Geoid Model dataset!");
+			" | Run install_geographiclib_dataset.sh script in order to install Geoid Model dataset!");
 		ros::shutdown();
 	}
 
-	// send static transform from "local_origin" (ENU) to "local_origin_ned" (NED)
-	publish_static_transform("local_origin", "local_origin_ned", Eigen::Affine3d(ftf::quaternion_from_rpy(M_PI, 0, M_PI_2)));
-	publish_static_transform("fcu", "fcu_frd", Eigen::Affine3d(ftf::quaternion_from_rpy(M_PI, 0, 0)));
+	// send static transform that are part of the tf tree as a vector
+	std::vector<geometry_msgs::TransformStamped> transformVector;
+	stack_static_transform("local_origin", "local_origin_ned", Eigen::Affine3d(ftf::quaternion_from_rpy(M_PI, 0, M_PI_2)),transformVector);
```

Comment:
Let's make this `odom` and `odom_ned` and add also `stack_static_transform("map", "map_ned", Eigen::Affine3d(ftf::quaternion_from_rpy(M_PI, 0, M_PI_2)),transformVector);`.

Reply:
How should odom be connected to map?

Reply:
They won't be connected. `odom` has an arbitrary origin and can drift over time while `map` has a fixed known origin and should not have drift.

Reply:
Actually, according to the REP 105, `map` can be the parent of `odom`. Just consider that everything that comes from the PX4 side in the `local_position` plugin should be set with `map` as the local frame and `base_link` as the body frame. What comes from the `odom` plugin should have `odom` as the local frame. Regarding the body frame, I guess we need to make clear that this body frame is "different" from the body frame on the data coming from the `local_position` plugin. So maybe `base_link_odom`?

Reply:
The body frames can be the same as long their is no tf transformation between odom and base_link